### PR TITLE
Add missing select types and refactor

### PR DESCRIPTION
**Pull request by JacopoPatroclo**
This PR aims to fix [this](https://github.com/kitajs/html/issues/177) issue

First I've added the missing properties to the select jsx element typescript definition. I've added `autocomplete` and I've noticed that other properties were declared by the FormEvents interface so I've extended the select definition with that interface to avoid duplication.

Second I've removed all the duplicated `undefined`. In this case, they should be equivalent to `?:`. Let me know @arthurfiorette if you see any issue with this and if I'm missing something.

Third I've added another test file that will be used in the future to check that all the types that this library provides are consistent with what is expected.

What still needs to be done:
- [ ] Write more tests related to the types
- [ ] Generate changelog (?)

### CODE REVIEW

**Comment by arthurfiorette** [comment]
> Second I've removed all the duplicated undefined. In this case, they should be equivalent to ?:. Let me know @arthurfiorette if you see any issue with this and if I'm missing something.

Using `?` and `undefined |` is required. Depending on the user's `exactOptionalPropertyTypes` configuration, it may force the user to define all properties as `<div prop={undefined} />`. 

In kita's tsconfig this is indeed duplicated, but we cannot ensure all users follow our tsconfig.

File: packages/html/jsx.d.ts [hunk 1]
```diff
@@ -29,21 +29,21 @@ declare namespace JSX {
   type CSSProperties = import('csstype').Properties;
 
   interface HtmlTag extends ElementChildrenAttribute, IntrinsicAttributes {
-    accesskey?: undefined | string;
```
Comment (arthurfiorette) [inline packages/html/jsx.d.ts:32]: please add `undefined |` again.

**Review by arthurfiorette — CHANGES_REQUESTED** [review]

**Comment by arthurfiorette** [comment]
To generate a changelog, run `pnpm changeset` in the project's root.

https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md

**Comment by arthurfiorette** [comment]
> Write more tests related to the types

We have two paths to follow here:

1. We find a complete test spec somewhere and copy/paste it here.
2. We wait for type issues to come and only merge them if they also add their tests here.

Adding more type tests without a issues we will probably end up with incorrect tests.

**Comment by JacopoPatroclo** [comment]
> > Write more tests related to the types
> 
> We have two paths to follow here:
> 
> 1. We find a complete test spec somewhere and copy/paste it here.
> 2. We wait for type issues to come and only merge them if they also add their tests here.
> 
> Adding more type tests without issues we will probably end up with incorrect tests.

That's a valid point. I'll invest a little bit of time looking for a standard spec that we could use.
If I can't find anything that fits; we should grow that particular test suite organically as you suggested (2)

**Comment by JacopoPatroclo** [comment]
@arthurfiorette I couldn't find any "standard" jsx definition for dom elements rendered on the server. There is a heavy focus (obviously) on client-side specific type definitions that are not good for the context of this library. We should roll our type definitions for now.

Ex. The onselect property is often defined as a function and not a string.

**Review by arthurfiorette — APPROVED** [review]

**Comment by arthurfiorette** [comment]
Awesome! As always, thanks for your work @JacopoPatroclo! Is this ready to merge?

**Comment by JacopoPatroclo** [comment]
> Awesome! As always, thanks for your work @JacopoPatroclo! Is this ready to merge?

I think so, maybe a squash merge is necessary to "hide" some "wrong" commits.

**Pull request merged**